src/compressor/Compressor.h

```diff
@@ -40,8 +40,19 @@ class Compressor {
 #ifdef HAVE_BROTLI
     COMP_ALG_BROTLI = 5,
 #endif
-    COMP_ALG_LAST	//the last value for range checks
+    COMP_ALG_LAST   //the last value for range checks
   };
+
+  // The names in this array must match the above values:
+  constexpr static char const* const compression_algorithm_names[] = {
```

Comment:
Better to define this as a std::map with explicit use of the COMP_* constants as the keys, rather than assuming those constants are sequential.

Reply:
Aaaaand I was afraid someone was going to say that. No objection, it just means more work. ;-)

Reply:
Another thing to point out here: the names are /also/ hardcoded elsewhere (in the configuration code), so even with this changing we're still in the unenviable position of requiring them to match up at a distance. (Maybe a good clean-up feature.)